pkg/receiver/telegrafreceiver/factory.go

```diff
@@ -47,6 +48,10 @@ func createDefaultConfig() config.Receiver {
 	return &Config{
 		ReceiverSettings: &rs,
 		SeparateField:    false,
+		// we mostly expect to get recoverable errors from the memory_limiter, which is unlikely to have
+		// a check interval less than 1s, so retrying much more frequently is pointless
+		ConsumeRetryDelay: 500 * time.Millisecond,
+		ConsumeMaxRetries: 10,
```

Comment:
Add those defaults to receiver's README

Reply:
I'm not sure this is a good idea. This is largely a technical detail that users should be able to ignore most of the time. It's only really configurable in case I'm wrong about the defaults in some specific circumstance.

Reply:
Well in this case it's a bit of an easter egg or security by obscurity :)

Not a huge fan of this approach.

Reply:
It's something that is only configurable just in case, and I'm somewhat concerned that putting it in the README will be more confusing than helpful. If someone is actually interested, they can read the source code and find out that way.

Reply:
Allright, the consensus is that we should document this with a caveat. Added to the README.